src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -706,6 +706,7 @@ export class SavedObjectsRepository {
     fields,
     namespaces,
     type,
+    typesAndNamespacesMap,
```

Comment:
`typesAndNamespacesMap` ends up bypassing the `_allowedTypes` check below. If I'm following this correctly, this would allow a consumer to retrieve types that were expressly forbidden by the client. I know the security wrapper is building this map itself based on the passed `type` and `namespaces`, but there's nothing that forces us to do that today, and nothing that prevents another consumer from (ab)using this differently in the future.

The implicit precedence of `typesAndNamespacesMap` over `type`/`namespaces` could lead to confusion too (even though you documented it). What do you think about having the repository throw an error if `typesAndNamespacesMap` is specified at the same time as `type` and/or `namespaces`? This would force consumers to be explicit about what they're asking for, and there would be less ambiguity around the passed options.

Reply:
> `typesAndNamespacesMap` ends up bypassing the `_allowedTypes` check below. If I'm following this correctly, this would allow a consumer to retrieve types that were expressly forbidden by the client.

Good catch... actually this is a bit worse, it would allow the consumer to retrieve hidden types (assuming they were somehow authorized to do so, *or* the Security plugin was disabled and they passed `typesAndNamespacesMap` into the server-side API).

> What do you think about having the repository throw an error if `typesAndNamespacesMap` is specified at the same time as `type` and/or `namespaces`? This would force consumers to be explicit about what they're asking for, and there would be less ambiguity around the passed options.

I love it! I'll make the change now.